ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/clientlib/js/product.js

```diff
@@ -52,14 +52,18 @@ class Product {
         price.currency = range.minimum_price.final_price.currency;
         price.regularPrice = range.minimum_price.regular_price.value;
         price.finalPrice = range.minimum_price.final_price.value;
-        price.discountAmount = range.minimum_price.discount.amount_off;
-        price.discountPercent = range.minimum_price.discount.percent_off;
+        if (range.minimum_price.discount) {
+            price.discountAmount = range.minimum_price.discount.amount_off;
```

Comment:
Not sure if it makes a different but with this approach `price.discountAmount` will be `undefined`, where setting it to `null` would be more explicit.

Reply:
It doesn't make any difference in this context. The variables are guarded by `price.discounted` so they will only be used if they evaluate to a truthy (!= null and != undefined) value.